### ISSUE TITLE
Kubernetes APIs are not cleaned up by Pulumi when destroying resources.

### ISSUE THREAD

**Issue by Lavender-Ding** [pulumi/pulumi-eks#67]
I‘m installing prometheus-operator and prometheus-adapter helm charts (https://github.com/helm/charts/tree/master/stable) via pulumi. A clean installation works good, but if i destroy these resources via Pulumi and create them again, it's complaining about some APIs already exist.

Here is my code:
`//Deploy the Prometheus Operator to the cluster
const prometheusOperator = new k8s.helm.v2.Chart("prometheus-operator", {
  repo: "stable",
  version: "3.0.0",
  namespace: "kube-system",
  chart: "prometheus-operator",
  transformations: [addKubeSystemNamespace],
  values: {
    alertmanager: {
      enabled: false,
    },
    grafana: {
      enabled: false,
    },
    defaultRules: {
      create: false,
    },
    coreDns: {
      enabled: false,
    },
    kubeEtcd: {
      enabled: false,
    },
    kubeScheduler: {
      enabled: false,
    },
    kubeControllerManager: {
      enabled: false,
    },
    "kube-state-metrics": {
      enabled: false,
      podSecurityPolicy: {
        enabled: false,
      },
    },
    prometheus: {
      service: {
        type: "LoadBalancer",
      },
    },
  },
}, {providers: {kubernetes: cluster.provider}});

// Deploy the Prometheus Adapter to the cluster
const prometheusAdapter = new k8s.helm.v2.Chart("prometheus-adapter", {
  repo: "stable",
  version: "",
  namespace: "kube-system",
  chart: "prometheus-adapter",
  transformations: [addKubeSystemNamespace],
  values: {
    prometheus: {
      url: "http://prometheus-operator-prometheus.kube-system.svc"
    },
  },
}, {providers: {kubernetes: cluster.provider}});`

And the error I'm getting everytime re-installing:
`Diagnostics:
  kubernetes:apiextensions.k8s.io:CustomResourceDefinition (kube-system/prometheuses.monitoring.coreos.com):
    error: Plan apply failed: customresourcedefinitions.apiextensions.k8s.io "prometheuses.monitoring.coreos.com" already exists

  kubernetes:apiextensions.k8s.io:CustomResourceDefinition (kube-system/alertmanagers.monitoring.coreos.com):
    error: Plan apply failed: customresourcedefinitions.apiextensions.k8s.io "alertmanagers.monitoring.coreos.com" already exists

kubernetes:apiregistration:APIService (kube-system/v1beta1.custom.metrics.k8s.io):
    error: Plan apply failed: apiservices.apiregistration.k8s.io "v1beta1.custom.metrics.k8s.io" already exists`

And if you do `kubectl api-versions` all these APIs are still there not being deleted.
It seems to me like Pulumi doesn't clean up APIs created completely when destroying resources. I don't have any issue when using kubectl to install/re-install.

**Comment by hausdorff** [comment]
@lblackstone not sure if you have the bandwidth to look at this, but FYI. LMK if we need to load balance

**Comment by lblackstone** [comment]
@hausdorff Yeah, I won't have time to look at this in M21. You can grab it, or we can punt to M22.

**Comment by lblackstone** [comment]
Spent a few minutes trying to reproduce, and it appears to be a bug related to the transformation on the charts (setting `kube-system` namespace). The diff logic isn't properly handling the non-namespaced resources (`CRD`s, `ClusterRoleBinding`s) since the namespace is being set in the transformation.
